src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs

```diff
@@ -255,20 +255,25 @@ internal partial class CodeFixService : ICodeFixService
             CodeActionRequestPriority? priority,
             CancellationToken cancellationToken)
         {
-            if (!(priority is null or CodeActionRequestPriority.Low)
-                || document is not Document sourceDocument)
-            {
+            if (!(priority is null or CodeActionRequestPriority.Low) || document is not Document sourceDocument)
+                return [];
+
+            var diagnostics = await document.GetCachedCopilotDiagnosticsAsync(cancellationToken).ConfigureAwait(false);
+            if (diagnostics.IsEmpty)
                 return [];
-            }
 
-            // Expand the fixable range for Copilot diagnostics to containing method.
-            // TODO: Share the below code with other places we compute containing method for Copilot analysis.
-            var root = await sourceDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var syntaxFacts = sourceDocument.GetRequiredLanguageService<ISyntaxFactsService>();
-            var containingMethod = syntaxFacts.GetContainingMethodDeclaration(root, range.Start, useFullSpan: false);
-            range = containingMethod?.Span ?? range;
+            if (sourceDocument.SupportsSyntaxTree)
```

Comment:
i'd just move this above, and bail out immediately.

Reply:
this is in case we want to support non cs/vb in roslyn workspace

Reply:
so, i'm wary about thsi currently as i don't understand the semantics.  specifically, what is hte impact of us passing in the full range of hte document into the methods you're calling.

Also, CodeFixService is firmly lang nutral.  Any lang specific checks (like getting syntax facts, or asking about things like "the containzing method") need to move into a lang-specific api.

That way a langauge like xaml could do the same sort of thing.  they need to be able to say "here's the span i want to limit myself to looking at".

Reply:
Make sense. How about we get this fix in to unblock insertion. I will follow up with a proper fix next.